### PR TITLE
TINKERPOP-1041

### PR DESCRIPTION
**Pull request by velo**
I decided to take a look into Tinkerpop-1041 and by doing so I end up setting up the project to build on a windows cloud CI and also fix a few problems I was able to spot on my machine.

This is not 100% done yet, just decided to give it some visibility before investing more time

### CODE REVIEW

File: gremlin-groovy/src/test/java/org/apache/tinkerpop/gremlin/groovy/engine/ScriptEnginesTest.java [hunk 1]
```diff
@@ -194,8 +195,8 @@ public void shouldFailUntilImportExecutes() throws Exception {
         threadEvalAndTriggerImport.join();
         threadImport.join();
 
-        assertTrue(successes.intValue() > 0);
-        assertTrue(failures.intValue() >= 500);
+        assertTrue("Success: " + successes.intValue() + " - Failures: " + failures.intValue(), successes.intValue() > 0);
```
Comment (spmallette) [inline gremlin-groovy/src/test/java/org/apache/tinkerpop/gremlin/groovy/engine/ScriptEnginesTest.java]: I tend to convert `assertTrue()` to hamcrest based asserts like `assertThat(success.intValue(), isGreaterThan(0))` which yields a better output on failure which i assume is what you are after here.

File: gremlin-test/src/main/java/org/apache/tinkerpop/gremlin/TestHelper.java [hunk 2]
```diff
@@ -50,6 +52,7 @@
 public final class TestHelper {
 
     private static final String SEP = File.separator;
+    private static final char URL_SEP = '/';
```
Comment (spmallette) [inline gremlin-test/src/main/java/org/apache/tinkerpop/gremlin/TestHelper.java]: ah! the class lookup is by URL so you need standard URL pathing to handle that (and not the windows file separator). cool

**Comment by spmallette** [comment]
Thanks for digging into this.  Can you please talk about what the "windows" maven profile is for?

I'd also like to know more about the appveyor config.  I guess you have that building against your fork with your appveyor account?

**Comment by spmallette** [comment]
Sorry - i also meant to mention - can you please also rename this PR to "TINKERPOP-1041" so it hooks into JIRA?

**Comment by velo** [comment]
the windows profile is to work around
https://issues.apache.org/jira/browse/HADOOP-10969
and
https://issues.apache.org/jira/browse/SPARK-6961

**Comment by velo** [comment]
BTW, @spmallette  can you enable appveyor on this project as well?
So we would know how does the build look like on windows
https://ci.appveyor.com/project/velo/incubator-tinkerpop

**Comment by spmallette** [comment]
@pluradj just thought i'd grab your attention on this issue....

@velo i can't just enable it as apache owns the repo - apache infra would have to enable it i guess.  we'll have to discuss on the mailing list anyway.  i'll bring it up

**Comment by velo** [comment]
@spmallette thanks

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  matrix:
+    - JAVA_HOME: C:\Program Files\Java\jdk1.8.0
+
+os: Windows Server 2012
+
+install:
+  - ps: |
+      Add-Type -AssemblyName System.IO.Compression.FileSystem
+      if (!(Test-Path -Path "C:\maven" )) {
+        (new-object System.Net.WebClient).DownloadFile('http://www.us.apache.org/dist/maven/maven-3/3.2.5/binaries/apache-maven-3.2.5-bin.zip', 'C:\maven-bin.zip')
```
Comment (steveloughran) [inline appveyor.yml]: https considered more secure

Comment (velo) [inline appveyor.yml]: https://www.us.apache.org/dist/maven/maven-3/3.3.9/binaries/apache-maven-3.3.9-bin.zip 
is considered an insecure connection

Comment (steveloughran) [inline appveyor.yml]: really? I'd grab it from ASF Central, or think about checking sha1 checksums (extra work)

Comment (spmallette) [inline appveyor.yml]: +1 for https however that would work

**Pull request closed without merging**